packages/schemas/src/workflow.ts

```diff
@@ -270,6 +271,76 @@ export class WorkflowManager {
     return step.config?.retries || 0;
   }
 
+  /**
+   * Inject a step dynamically at runtime (not persisted to YAML)
+   * @param step - The step to inject
+   * @param position - Where to inject: 'before' (start) or 'after' (end)
+   * @param referenceStepId - Optional step ID to position relative to
+   */
+  injectStep(
+    step: WorkflowStep,
+    position: 'before' | 'after' = 'before',
+    referenceStepId?: string
+  ): void {
```

Comment:
I like this logic! I think it could be pretty useful to add extra steps that we consider will improve the result.